how-to/misc/identity-provider-enablement.rst

```diff
@@ -0,0 +1,354 @@
+    +-----------------------------+------------+----------+
+    | Name                        | Provider   | Protocol |
+    +-----------------------------+------------+----------+
+    | Keystone Credentials        | Built-in   | keystone |
+    | canonical-identity-platform | canonical  | openid   |
+    | my-google-idp               | google     | openid   |
+    +-----------------------------+------------+----------+
+
+Okta and Entra ID have identical procedures.
+
+Make a note of the name of the provider and of the protocol. We will use them in the next steps to enable these providers in keystone.
+
+Note, you should already see them in `Horizon`, but you will only be able to use them after we've mapped them to domains and projects. Examples bellow.
+
+Removing a provider
+~~~~~~~~~~~~~~~~~~~
+
+Removing a provider is a matter of running:
+
+::
+
+    sunbeam identity provider remove my-google-idp --yes-i-mean-it
+
+
+Note, this will not remove any resources created by the cloud administrator using the `openstack` command.
+
+Making use of the new providers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that we've made the providers available to the cloud, we can enable them in keystone, map them to a domain and create rules on how users should
+be mapped to projects.
+
+You can create a new domain or you can use an existing domain to map it to the IDP. For the purposes of this guide, we'll create a new one:
+
+::
+
+    openstack domain create \
+        --description="Federated Google domain" \
+        google
+
+Get the issuer URL for the desired IDP. In this case we'll go with `my-google-idp` from the output above:
+
+::
+
+    ISSUER_URL=$(sunbeam identity provider list \
+        --format=yaml |  yq -r '."my-google-idp".issuer_url')
+
+Create the identity provider in Keystone:
+
+::
+
+    openstack identity provider create \
+        --remote-id $ISSUER_URL \
+        --domain google \
+        my-google-idp
+
+
+Note, the name of the identity provider must match the name in the table outputted by sunbeam.
+
+Create a group which we will assign to federated users:
+
+::
+
+    openstack group create federated_users \
+        --domain google
+
+Create a project. You can of course name something else:
+
+::
+
+    openstack project create \
+        --domain google \
+        federated_project
+
+Add a role for the group on the project we want to use:
+
+::
+
+    openstack role add \
+        --group federated_users \
+        --project federated_project \
+        --group-domain google \
+        --project-domain google \
+        member
+
+
+Next, we need to create some mapping rules between the remote users that come in from the IDP and local openstack users. The rules instruct Keystone on how
+to automatically create local users and to potentially assign them to groups, projects, domains, etc. You may consult `the official documentation <https://docs.openstack.org/keystone/latest/admin/federation/mapping_combinations.html>`_
```

Comment:
s/on how to/how to/

Also, I'd remove `potentially` from the sentence. The reference to the official keystone documentation will help provide more of the details here.